src/rp2_common/pico_cyw43_driver/btstack_hci_transport_cyw43.c

```diff
@@ -21,6 +21,9 @@
 #error HCI_ACL_CHUNK_SIZE_ALIGNMENT not defined or not a multiply of 4. Please update btstack_config.h
 #endif
 
+// assert buffer for cyw43_bluetooth_hci_read starts word aligned (word align pre buffer)
+#define HCI_INCOMING_PRE_BUFFER_SIZE_ALIGNED ((HCI_INCOMING_PRE_BUFFER_SIZE + 3) & 0xfffc)
```

Comment:
this is mildly confusing - why 0xfffc rather than ~3 - does this say something about this always being <64K?

Reply:
I think ~3 would do the job. It's the cyw43 chip that needs the buffers to be aligned I believe.

Reply:
Yes, ~3 would be fine. The HCI Buffer matches the largest packet over the air, which is 1023. For reasons, BNEP (tethering) puts Ethernet frames (up to 1651 or something linke that) into an L2CAP packet that we then put into an large HCI buffer. So 1659 is sufficient for all use cases.

I was told that the cyw43 driver requires its read/write buffer to be word aligned.
 
I'll try to add some more comments on the data. The 4 bytes for the cyw43 specific header could be subsumed in the general HCI_INCOMING_PRE_BUFFER_SIZE, but that would increase the logic even more (but save a full 4 bytes :).

Reply:
yeah; i don't think the 4 bytes is critical, just wanted to reason as to whether it was correct ;-)